pkg/nsx/util/utils.go

```diff
@@ -553,46 +552,16 @@ func FindTag(tags []model.Tag, tagScope string) string {
 }
 
 func CasttoPointer(obj interface{}) interface{} {
-	switch v := obj.(type) {
-	case mpmodel.PrincipalIdentity:
-		return &v
-	case model.Rule:
-		return &v
-	case model.StaticRoutes:
-		return &v
-	case model.HostTransportNode:
-		return &v
-	case model.ClusterControlPlane:
-		return &v
-	case model.IpAddressPool:
-		return &v
-	case model.GenericPolicyRealizedResource:
-		return &v
-	case model.Vpc:
-		return &v
-	case model.LBService:
-		return &v
-	case model.IpAddressPoolBlockSubnet:
-		return &v
-	case model.Group:
-		return &v
-	case model.SecurityPolicy:
-		return &v
-	case model.Share:
-		return &v
-	case model.SegmentPort:
-		return &v
-	case model.VpcSubnet:
-		return &v
-	case model.VpcSubnetPort:
-		return &v
-	case model.IpAddressBlock:
-		return &v
-	default:
-		objType := reflect.TypeOf(obj)
-		log.Info("Unsupported type", "objType", objType)
+	if obj == nil {
 		return nil
 	}
+	if reflect.TypeOf(obj).Kind() == reflect.Ptr {
+		return obj
+	}
+
+	pointer := reflect.New(reflect.TypeOf(obj))
+	pointer.Elem().Set(reflect.ValueOf(obj))
+	return pointer.Interface()
```

Comment:
Go 1.18 and later versions support generics, I think it will make this function more type-safe and reduce its reliance on reflect.

Reply:
Maybe you can check `k8s.io/utils/ptr` to simplify this code.